src/pages/products/main/cardContainer.js

```diff
@@ -15,15 +16,17 @@ function sastaLoader() {
 }
 
 function CardContainer() {
+  const [getSearchedProducts] = useSearchFeature();
   const { filteredProducts } = useProducts();
+  const products = getSearchedProducts(filteredProducts);
 
   return (
     <div className="products-cards-container">
-      {filteredProducts &&
-        filteredProducts.map((product) => (
+      {products &&
```

Comment:
suggestion: check like products?.length > 0 instead of products

Reply:
Suggestion Adopted.